src/main/java/project/task/TaskList.java

```diff
@@ -0,0 +1,118 @@
+package project.task;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a list of {@code Task}s in the application.
+ * Encapsulates storage and operations on {@code Task}s.
+ */
+public class TaskList {
+    private List<Task> tasks;
```

Comment:
I think this is trivial but perhaps you could initialize the variable tasks under class TaskList to new ArrayList()? Then you might be able to do away with the empty constructor.

Reply:
That's true, thanks for the suggestion! I'll make the necessary changes 👍